vispy/visuals/markers.py

```diff
@@ -27,95 +28,170 @@
-    float edgewidth = max(v_edgewidth, 1.0);
-    gl_PointSize = ($v_size) + 4.*(edgewidth + 1.5*v_antialias);
+
+    vec4 pos = vec4(a_position, 1);
+    vec4 fb_pos = $visual_to_framebuffer(pos);
+    gl_Position = $framebuffer_to_render(fb_pos);
+
+    // NOTE: gl_stuff uses framebuffer coords!
+
+    if (u_scaling == true) {
+        // calculate point size from visual to framebuffer coords to determine size
+        vec4 x = $framebuffer_to_visual(fb_pos + vec4(big_float, 0, 0, 0));
+        x = (x - pos);
+        vec4 size_vec = $visual_to_framebuffer(pos + normalize(x) * a_size);
+        $v_size = size_vec.x / size_vec.w - fb_pos.x / fb_pos.w;
+        v_edgewidth = ($v_size / a_size) * a_edgewidth;
+    }
+    else {
+        $v_size = a_size * u_px_scale;
+        v_edgewidth = a_edgewidth * u_px_scale;
+    }
+
+    // gl_PointSize is the diameter
+    gl_PointSize = $v_size + 4. * (v_edgewidth + 1.5 * u_antialias);
+
+    if (u_sphere == true) {
+        // Get the framebuffer z direction relative to this sphere in visual coords
+        vec4 z = $framebuffer_to_visual(fb_pos + vec4(0, 0, big_float, 0));
+        z = (z - pos);
+        // Get the depth of the sphere in its middle point on the screen
+        // size/2 because we need the radius, not the diameter
+        vec4 depth_z_vec = $visual_to_framebuffer(pos + normalize(z) * a_size / 2);
+        v_depth_middle = depth_z_vec.z / depth_z_vec.w - fb_pos.z / fb_pos.w;
+        // size ratio between aliased and non-aliased, needed for correct depth
+        v_alias_ratio = gl_PointSize / $v_size;
+    }
 }
 """
 
 
 frag = """#version 120
+uniform vec3 u_light_position;
+uniform vec3 u_light_color;
+uniform float u_light_ambient;
+uniform float u_alpha;
+uniform float u_antialias;
+uniform bool u_sphere;
+
 varying vec4 v_fg_color;
 varying vec4 v_bg_color;
 varying float v_edgewidth;
-varying float v_antialias;
+varying float v_depth_middle;
+varying float v_alias_ratio;
 
 void main()
 {
     // Discard plotting marker body and edge if zero-size
     if ($v_size <= 0.)
         discard;
 
-    float edgewidth = max(v_edgewidth, 1.0);
     float edgealphafactor = min(v_edgewidth, 1.0);
 
-    float size = $v_size + 4.*(edgewidth + 1.5*v_antialias);
+    float size = $v_size + 4.*(v_edgewidth + 1.5*u_antialias);
     // factor 6 for acute edge angles that need room as for star marker
 
     // The marker function needs to be linked with this shader
     float r = $marker(gl_PointCoord, size);
 
     // it takes into account an antialising layer
-    // of size v_antialias inside the edge
+    // of size u_antialias inside the edge
     // r:
     // [-e/2-a, -e/2+a] antialising face-edge
     // [-e/2+a, e/2-a] core edge (center 0, diameter e-2a = 2t)
     // [e/2-a, e/2+a] antialising edge-background
-    float t = 0.5*v_edgewidth - v_antialias;
+    // use max because we don't want negative transition zone
+    float t = max(0.5*v_edgewidth - u_antialias, 0);
```

Comment:
Actually, removing the edge completely is impossible in master, but I changed it here.